mix.exs

```diff
@@ -21,7 +21,8 @@ defmodule Bamboo.Mixfile do
       start_permanent: Mix.env() == :prod,
       package: package(),
       docs: [main: "readme", extras: ["README.md"]],
-      deps: deps()
+      deps: deps(),
+      xref: [exclude: [IEx, ExMachina]]
```

Comment:
Hi @tomciopp, 

Thanks so much for all the extra information you gave regarding the dependencies. I was looking through this PR again, and this caught my eye. Could you give some extra info as to why we're excluding these from xref? 

When I read [the Elixir 1.11 announcement](https://elixir-lang.org/blog/2020/10/06/elixir-v1-11-0-released/#compiler-checks-application-boundaries), it seems like these two should not be excluded here. 

It seems like `iex` should fall under the "part of Elixir" portion of libraries, so we should put it in `extra_applications`, right? Or to dig even further, do we even need to list `IEx` at all? Where does Bamboo use `IEx`? 

And `ExMachina` is defined in `deps`, so it seems like we shouldn't need to list it here either, right?

Thanks for all the info on the rest of the PR. I think this is the last question I have.

Reply:
If I'm not mistaken, adding `:iex` to `:extra_applications` would imply that IEx should be started whenever Bamboo starts, which I don't think is the desired behavior.

As for ExMachina, I'm not sure why that needed to be excluded.

Reply:
IEx is used in the following file
`lib/mix/start_sent_email_viewer_task.ex:70`

Here is the warning emitted

```
warning: IEx.started?/0 defined in application :iex is used by the current application but the current application does not directly depend on :iex. To fix this, you must do one of:

  1. If :iex is part of Erlang/Elixir, you must include it under :extra_applications inside "def application" in your mix.exs

  2. If :iex is a dependency, make sure it is listed under "def deps" in your mix.exs

  3. In case you don't want to add a requirement to :iex, you may optionally skip this warning by adding [xref: [exclude: IEx]] to your "def project" in mix.exs
```

We don't want IEx to be started when Bamboo starts so we use option 3 to exclude it.

Reply:
There is a separate warning with ExMachina, but I believe it is related to the library itself. I will remove the module from the xref exclude list. I've reproduced the warning below for your convenience.

```
warning: incompatible types:

    map() !~ Bamboo.NormalizeAndPushStrategy

in expression:

    # test/support/normalize_and_push_strategy.ex:2
    custom_strategy_module.function_name

where "custom_strategy_module" was given the type Bamboo.NormalizeAndPushStrategy in:

    # test/support/normalize_and_push_strategy.ex:2
    custom_strategy_module = Bamboo.NormalizeAndPushStrategy

where "custom_strategy_module" was given the type map() (due to calling var.field) in:

    # test/support/normalize_and_push_strategy.ex:2
    custom_strategy_module.function_name

HINT: "var.field" (without parentheses) implies "var" is a map() while "var.fun()" (with parentheses) implies "var" is an atom()

Conflict found at
  test/support/normalize_and_push_strategy.ex:2: Bamboo.NormalizeAndPushStrategy.__using__/1
```